src/index.js

```diff
@@ -1,224 +1,566 @@
-process.env.SENTRY_DSN =
-  process.env.SENTRY_DSN ||
-  'https://c76a294b95744d2cbd8bbf52cb13cd5f:7d6b141aa5fd41b882cf12088683b30a@sentry.cozycloud.cc/51'
-
-const {
-  BaseKonnector,
-  requestFactory,
-  scrape,
-  log
-} = require('cozy-konnector-libs')
-
-let request = requestFactory()
-const j = request.jar()
-request = requestFactory({
-  // debug: true,
-  cheerio: true,
-  json: false,
-  jar: j,
-  insecureHTTPParser: true, // cdiscount sends bad headers sometimes which are not recognized by node
-  headers: {
-    Referer: 'https://clients.cdiscount.com/account/home.html'
-  }
-})
-
-module.exports = new BaseKonnector(start)
-
-const vendor = 'cdiscount'
-const baseurl = 'https://clients.cdiscount.com'
-
-async function start(fields) {
-  const challengeToken = await fetchChallengeCookie(
-    `${baseurl}/Account/LoginLight.html?referrer=`
-  )
-  j.setCookie(challengeToken, 'https://order.cdiscount.com')
-
-  log('info', 'Authenticating ...')
-  await authenticate.bind(this)(fields.login, fields.password)
-  log('info', 'Successfully logged in')
-
-  log('info', 'Fetching list of orders')
-  const saleFolderIDs = await getSaleFolderIDs(challengeToken)
-  log('info', 'Fetching orders')
-  const orders = await fetchAllOrders(saleFolderIDs, challengeToken)
-  log('info', 'Fetching bills')
-  const bills = await fetchBills(orders)
-  log('info', 'Saving data to Cozy')
-  await this.saveBills(bills, fields, {
-    linkBankOperations: false,
-    fileIdAttributes: ['vendorRef']
-  })
-}
+import { ContentScript } from 'cozy-clisk/dist/contentscript'
+import Minilog from '@cozy/minilog'
+import waitFor from 'p-wait-for'
+import { parse, format } from 'date-fns'
+import { fr } from 'date-fns/locale'
+const log = Minilog('ContentScript')
+Minilog.enable('cdiscountCCC')
+
+// const baseUrl = 'https://cdiscount.com'
+const loginFormUrl =
+  'https://order.cdiscount.com/Phone/Account/LoginLight.html?referrer='
+
+class CdiscountContentScript extends ContentScript {
+  onWorkerReady() {
+    window.addEventListener('DOMContentLoaded', () => {
```

Comment:
Are you sure DOMContentLoaded is always triggered after onWorkerReady (or else credentials won't be saved)

Reply:
If i remember correctly we used to have issues when we were not waiting for the DOMContentLoaded, so I'll assume yes